packages/@sanity/base/src/components/WithReferringDocuments.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import documentStore from 'part:@sanity/base/datastore/document'
+import {withPropsStream} from 'react-props-stream'
+import {concat, of} from 'rxjs'
+import {distinctUntilChanged, map, switchMap} from 'rxjs/operators'
+
+const loadProps = receivedProps$ =>
+  receivedProps$.pipe(
+    distinctUntilChanged((prev, next) => prev.id === next.id),
+    switchMap(receivedProps =>
+      concat(
+        of({...receivedProps, referringDocuments: [], isLoading: true}),
+        documentStore.query('*[references($docId)] [0...101]', {docId: receivedProps.id}).pipe(
+          map(event => ({
+            ...receivedProps,
+            referringDocuments: event.documents,
+            isLoading: false
+          }))
+        )
+      )
+    )
+  )
+
+export const WithReferringDocuments = withPropsStream(loadProps, ({children, ...props}) =>
+  children(props)
+)
+
+export function enhanceWithReferringDocuments(Component) {
+  function Wrapper(props) {
+    return (
+      <WithReferringDocuments id={props.published._id}>
+        {({isLoading, referringDocuments}) => (
+          <Component
+            {...props}
+            referringDocuments={referringDocuments}
+            isCheckingReferringDocuments={isLoading}
+          />
+        )}
+      </WithReferringDocuments>
+    )
+  }
+  Wrapper.displayName = `enhanceWithReferringDocuments(${Component.displayName || Component.name})`
+  Wrapper.propTypes = {
+    published: PropTypes.object
```

Comment:
Ah, good catch. This is leftover from how it was implemented in desk tool - where it was used to wrap components like `<ConfirmDelete>`, which already got passed the `published` and `draft` documents. To avoid ending up with having to pass both `id`, `published` and `draft` which makes the usage site a bit confusing, (at least in my opinion):

```jsx
<ConfirmDelete
  draft={draft}
  published={published}
  id={published._id} <-- this is needed by the HOC now
/>
```

This is a general problem with HOCs, and I kind of want to get rid of the `enhanceWithReferringDocuments` altogether, so moved it into desk tool to preserved the old behavior for now.